pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -76,21 +71,22 @@ func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context,
 	return nil
 }
 
-func (dv *dynamic) validateClusterMSI(ctx context.Context, oc *api.OpenShiftCluster, roleDefinitions armauthorization.RoleDefinitionsClient) error {
-	for resourceID, identity := range oc.Identity.UserAssignedIdentities {
+func (dv *dynamic) ValidateClusterUserAssignedIdentity(ctx context.Context, oc *api.OpenShiftCluster, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	dv.log.Print("ValidateClusterUserAssignedIdentity")
+
+	for resourceID := range oc.Identity.UserAssignedIdentities {
 		_, err := azure.ParseResourceID(resourceID)
 		if err != nil {
 			return err
 		}
-
-		return dv.validateClusterMSIPermissions(ctx, identity.PrincipalID, oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities, roleDefinitions)
+		return dv.validateClusterMSIPermissions(ctx, oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities, roleDefinitions)
```

Comment:
How are we validating the cluster MSI if we parse the MSI resource ID and then do nothing with it?

Reply:
Ohh thanks, previously we needed the principalID so we were iterating over the oc.Identity.UserAssignedIdentites. It is no more needed. We can call the validateClusterMSIPermissions directly.


> My reading of the new code is that changes are being made not to the platform identity validation but to the cluster MSI validation, and that the cluster MSI is no longer being validated. I find check access v2 confusing in general though. Could you help me understand it a bit better?

So for the Cluster MSI we have the access token(i.e. azcore.TokenCredential), so we just use the exact similar CheckAccessV2 flow  as used by FPSP and CSP. Getting OID and AccessToken from the azcore.TokenCredential

But since we don't get credential for the Platform Workload Identity, we just pass OID adding an else case for creating AuthorizationRequest only based on OID and not using claims from the access token here:-

Reply:
Got it, thank you for humoring my questions!